generator/internal/discovery_to_proto.cc

```diff
@@ -298,7 +298,7 @@ std::set<std::string> FindAllRefValues(nlohmann::json const& json) {
   } else if (json.contains("additionalProperties") || json.contains("items")) {
     fields = json;
   } else {
-    return {};
+    return ref_values;
```

Comment:
I would suggest dropping the `else` altogether, and allowing an empty `fields` to return an empty `ref_values` without a special case.